addons/pkg-npm/model-java/src/main/java/org/commonjava/indy/pkg/npm/model/PackageMetadata.java

```diff
@@ -446,7 +446,7 @@ public boolean merge( PackageMetadata source )
         for ( final String v : sourceVersions.keySet() )
         {
             VersionMetadata value = sourceVersions.get( v );
-            if ( !versions.containsKey( v ) )
+            if ( !isForGroup || !versions.containsKey( v ) )
```

Comment:
What does this do? Maybe it's just too late, but I can't quite understand what happens when you're merging to a group metadata file that's different here.

Reply:
@jdcasey for the group merging, same version metadata in the versions will not be coverd, because group only accepts the first one coming to the group.

While for the publish merging, it's different, publish should always accept the latest updated version metadta (no matter the version is existed or not)

So this judgement tells: if this is 'for publish merging' or 'not contains the version for group merging', versions will be updated.

Reply:
Thanks!